### PR TITLE
[microcloud] Added new plugin support for MicroCloud.

### PR DESCRIPTION
**Pull request by alanbach**
This implements a new small plugin support for the MicroCloud snap.

### CODE REVIEW

File: sos/report/plugins/microcloud.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    commands.
+    """
+
+    short_desc = 'MicroCloud Snap'
+    plugin_name = "microcloud"
+    profiles = ('container',)
+
+    packages = ('microcloud',)
+
+    microcloud_cmd = "microcloud"
```
Comment (pmoravec) [inline sos/report/plugins/microcloud.py:28]: You don't use the variable either where.

File: sos/report/plugins/microcloud.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        self.add_journal(units="snap.microcloud.*")
+
+        microcloud_subcmds = [
+            'cluster list',
+            '--version'
+        ]
+        self.add_copy_spec([
+            '/var/snap/microcloud/common/state/database/cluster.yaml',
+            '/var/snap/microcloud/common/state/database/info.yaml',
+        ])
```
Comment (pmoravec) [inline sos/report/plugins/microcloud.py:40]: I guess there are no secrets there (like passwords to the database..?)?

File: sos/report/plugins/microcloud.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+        self.add_cmd_output([
+            f"microcloud {subcmd}" for subcmd in microcloud_subcmds
+        ])
```
Comment (pmoravec) [inline sos/report/plugins/microcloud.py:44]: As you will respin the PR, could you please add here the `vim` config for indentation? The `# vim: set et ts=4 sw=4 :` line like e.g. https://github.com/sosreport/sos/blob/main/sos/report/plugins/abrt.py#L44 or at the botton of any other source code file.

Comment (arif-ali) [reply sos/report/plugins/microcloud.py:44]: > As you will respin the PR, could you please add here the `vim` config for indentation? The `# vim: set et ts=4 sw=4 :` line like e.g. https://github.com/sosreport/sos/blob/main/sos/report/plugins/abrt.py#L44 or at the botton of any other source code file.

we ought to add that to the Contribution Guidelines as well for future contributors

File: sos/report/plugins/microcloud.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
+# Copyright (C) 2024 Alan Baghumian <alan.baghumian@canonical.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
```
Comment (arif-ali) [inline sos/report/plugins/microcloud.py:7]: We're just changing the guidelines on licensing, so would you update this to `version 2 or later` please

File: sos/report/plugins/microcloud.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    short_desc = 'MicroCloud Snap'
+    plugin_name = "microcloud"
+    profiles = ('container',)
+
+    packages = ('microcloud',)
+
+    microcloud_cmd = "microcloud"
+
+    def setup(self):
+        self.add_journal(units="snap.microcloud.*")
```
Comment (arif-ali) [inline sos/report/plugins/microcloud.py:31]: do we need to check for service status as well?

File: sos/report/plugins/microcloud.py (hunk 1 above)
Comment (alanbach) [reply sos/report/plugins/microcloud.py:28]: Yup, removing that last one, we do need the packages one thought

File: sos/report/plugins/microcloud.py (hunk 3 above)
Comment (alanbach) [reply sos/report/plugins/microcloud.py:44]: Done! Coming.

File: sos/report/plugins/microcloud.py (hunk 4 above)
Comment (alanbach) [reply sos/report/plugins/microcloud.py:7]: Done. Coming.

File: sos/report/plugins/microcloud.py (hunk 5 above)
Comment (alanbach) [reply sos/report/plugins/microcloud.py:31]: Nah, the systemd module is already collecting it, so no need for a duplicated efforts

**Review by pmoravec — APPROVED** [review]

**Review by arif-ali — APPROVED** [review]

**Pull request merged**